internal/jimm/bootstrap/bootstrap.go

```diff
@@ -265,18 +307,26 @@ func (b *bootstrapManager) BootstrapJob(
 			return errors.E(fmt.Errorf("failed to check if controller exists: %w", err))
 		}
 
+		// TODO(ale8k): When downloading, it takes a while and the CLI has no output. Download progress would be a VERY nice to have here.
 		binary, err := b.binaryStore.Get(
 			ctx,
 			jujuclistore.JujuBinarySpec{
 				Version: p.CLIVersion,
-				Os:      p.CLIVersion,
-				Arch:    p.CLIArch,
+				// This is a best effort. The launchpad URL just so happens to have similar filenames to runtime.GOOS and runtime.GOARCH.
+				// If this ever changes, we should update the binary store to use the correct URL
+				// or we should detect the OS and arch here. We don't want to detect it in the binary store
+				// because the consumer may want to detect it in different ways.
+				Os:   runtime.GOOS,
+				Arch: runtime.GOARCH,
 			},
 		)
 		if err != nil {
 			return errors.E(fmt.Errorf("failed to get Juju binary: %w", err))
 		}
-
+		// TODO(ale8k): Is it possible to tell the client to only attempt to get status after binary is downloaded?
+		// Because as it stands, the client will be unncessarily polling (perhaps we could put download progress in the logs?).
+		// That way the status calls wouldn't be unncessary and serve a purpose.
```

Comment:
We could have the client print a "Waiting for bootstrap to start" message at the beginning until logs start coming in.

Reply:
It does already print a pending job message, think that'd suffice?

Reply:
Removed comment